RIOT-OS/RIOT
# gnrc_lorawan: refactor

### Contribution description
This PR:

- Removes the dependency to `gnrc_pktbuf` in GNRC LoRaWAN. Besides reducing the ROM usage a lot, this allows the usage of GNRC LoRaWAN without depending on GNRC specific stuff
- Rename radio events functions to ease maintaining
- Some minor fixes wit "BUSY" logic
- Remove dependency to `gnrc_neterr` (now `gnrc_lorawan` is fully asynchronous)
- Uncrustifies GNRC LoRaWAN files

This step is required in order to add missing class B and class C support.



### Testing procedure
This should be tested with care.
I suggest trying the following:
- Join a LoRaWAN network using ABP and OTAA
- Send some uplinks and downlinks (to port 2). Check that data is received in both directions and that there aren't leaks in the pktbuf (use `gnrc_pktbuf_cmd`module).
- Try sending several packets in a row. It shouldn't crash and the MAC layer should just drop a packet if the MAC is busy.





### Issues/PRs references
~~Depends on #14058~~
